lib/server/versionFallback.js

```diff
@@ -194,7 +195,27 @@ function diffLatestDoc(file, id) {
 // the version of the file to be used, and its language
 function processVersionMetadata(file, version, useVersion, language) {
   const metadata = extractMetadata(fs.readFileSync(file, 'utf8')).metadata;
-  metadata.source = 'version-' + useVersion + '/' + path.basename(file);
+
+  // Add subdirectory information to versioned_doc metadata
+  // Example: `versioned_docs/version-1.1.6/projectA/readme.md` file with id `version-1.1.6-readme`
+  // and original_id `readme` will have metadata id of `version-1.1.6-projectA/readme` and original_id `projectA/readme`
+  const subDir = utils.getSubDir(
+    file,
+    path.join(CWD, 'versioned_docs', `version-${useVersion}`)
+  );
+  if (subDir) {
+    metadata.original_id = subDir + '/' + metadata.original_id;
+    metadata.id = metadata.id.replace(
```

Comment:
Could you use template strings instead of concatenating the strings? It'll be easier to read.